### ISSUE TITLE
Poor results on validation dataset

### ISSUE THREAD

**Issue by mansi-aggarwal-2504** [matterport/Mask_RCNN#2652]
I have a question. Does mask rcnn not adjust its weights and learning on the basis of validation dataset after each epoch. Like I have a a dataset divided into train, val and test. Train and val are supplied for training. And if I run the the model on validation dataset, the results are quite poor itself, let alone test dataset. This means validation dataset is not used for training? Just for us to check our val score while training is going on?

**Comment by agbozo1** [comment]
Facing a similar situation. Did you augment during training?
Or tweak the training epochs / steps per epoch.

**Comment by mansi-aggarwal-2504** [comment]
Hello @agbozo1 
I tried augmentations. 
```
augmentation = iaa.SomeOf((0, None), [
iaa.Fliplr(1),
iaa.Flipud(1),
iaa.OneOf([iaa.Affine(rotate = 90 * i) for i in range(1, 3)]),
iaa.OneOf([iaa.Affine(rotate = -90 * i) for i in range(1, 3) ]),
iaa.Affine(scale={"x": (0.5, 1.5), "y": (0.5, 1.5)}),
iaa.sizelib.CropAndPad(percent=([0.05, 0.1], [0.05, 0.1], [0.05, 0.1], [0.05, 0.1]))
])
```

I am running the model on 70 epochs and 50 STEPS_PER_EPOCH. Without augmentation, the validation score wouldn't improve after 40 epochs but once I applied augmentations, the score wouldn't improve just after 30 epochs. I am not sure what to do to get better results.

**Comment by AndreEstevam** [comment]
The validation set is for monitoring purposes only, it does not affect the weights of the model.

**Comment by agbozo1** [comment]
> The validation set is for monitoring purposes only, it does not affect the weights of the model.

@AndreEstevam what is a solid training strategy you would apply for instance segmentation?
